keras/applications/densenet.py

```diff
@@ -0,0 +1,341 @@
+                weights_path = get_file(
+                    'densenet201_weights_tf_dim_ordering_tf_kernels.h5',
+                    DENSENET201_WEIGHT_PATH,
+                    cache_subdir='models',
+                    file_hash='7bb75edd58cb43163be7e0005fbe95ef')
+        else:
+            if blocks == [6, 12, 24, 16]:
+                weights_path = get_file(
+                    'densenet121_weights_tf_dim_ordering_tf_kernels_notop.h5',
+                    DENSENET121_WEIGHT_PATH_NO_TOP,
+                    cache_subdir='models',
+                    file_hash='4912a53fbd2a69346e7f2c0b5ec8c6d3')
+            elif blocks == [6, 12, 32, 32]:
+                weights_path = get_file(
+                    'densenet169_weights_tf_dim_ordering_tf_kernels_notop.h5',
+                    DENSENET169_WEIGHT_PATH_NO_TOP,
+                    cache_subdir='models',
+                    file_hash='50662582284e4cf834ce40ab4dfa58c6')
+            elif blocks == [6, 12, 48, 32]:
+                weights_path = get_file(
+                    'densenet201_weights_tf_dim_ordering_tf_kernels_notop.h5',
+                    DENSENET201_WEIGHT_PATH_NO_TOP,
+                    cache_subdir='models',
+                    file_hash='1c2de60ee40562448dbac34a0737e798')
+        model.load_weights(weights_path)
+    elif weights is not None:
+        model.load_weights(weights)
+
+    return model
+
+
+def DenseNet121(include_top=True,
+                weights='imagenet',
+                input_tensor=None,
+                input_shape=None,
+                pooling=None,
+                classes=1000):
+    return DenseNet([6, 12, 24, 16],
+                    include_top, weights,
+                    input_tensor, input_shape,
+                    pooling, classes)
+
+
+def DenseNet169(include_top=True,
+                weights='imagenet',
+                input_tensor=None,
+                input_shape=None,
+                pooling=None,
+                classes=1000):
+    return DenseNet([6, 12, 32, 32],
+                    include_top, weights,
+                    input_tensor, input_shape,
+                    pooling, classes)
+
+
+def DenseNet201(include_top=True,
+                weights='imagenet',
+                input_tensor=None,
+                input_shape=None,
+                pooling=None,
+                classes=1000):
+    return DenseNet([6, 12, 48, 32],
+                    include_top, weights,
+                    input_tensor, input_shape,
+                    pooling, classes)
+
+
+def preprocess_input(x):
+    """Preprocesses a numpy array encoding a batch of images.
+
+    # Arguments
+        x: a 4D numpy array consists of RGB values within [0, 255].
+
+    # Returns
+        Preprocessed array.
+    """
+    x /= 255.
+    x[:, :, :, 0] -= 0.485
+    x[:, :, :, 1] -= 0.456
+    x[:, :, :, 2] -= 0.406
+    x[:, :, :, 0] /= 0.229
+    x[:, :, :, 1] /= 0.224
+    x[:, :, :, 2] /= 0.225
```

Comment:
This preprocessing is only implemented for `channels_last` data format. Please add support for both.